Android/src/main/java/applicaster/analytics/firebase/FirebaseAgent.java

```diff
@@ -254,6 +263,13 @@ public void sendUserProperties(JSONObject params) throws JSONException {
         super.sendUserProperties(params);
         if (params != null && isSendUserData) {
 
+            if (allowTrackingUserId()) {
+                String storedUserId = getLocalStorageUserId();
+                if (StringUtil.isNotEmpty(storedUserId)) {
+                    params.put("user_id", storedUserId);
```

Comment:
You should use the method:
setUserId();
https://firebase.google.com/docs/analytics/userid#android